sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceManager.scala

```diff
@@ -99,7 +99,7 @@ object DataSourceManager extends Logging {
 
   private def normalize(name: String): String = name.toLowerCase(Locale.ROOT)
 
-  private def initialStaticDataSourceBuilders: Map[String, UserDefinedPythonDataSource] = {
+  private lazy val initialStaticDataSourceBuilders: Map[String, UserDefinedPythonDataSource] = {
```

Comment:
Just a question. How many times is this invoked currently?

Reply:
For now, it happens every session creation. Yeah, let me file a JIRA